### DISCUSSION TITLE
how to change default behavior of workspace 2?

### DISCUSSION THREAD

**Discussion by kronikpillow** [hyprwm/Hyprland#987] [Q&A]
hi there, loving the project so far, recently switched from dwm which I'v been using exclusively for 5 years, didn't even try any other wm's, but hyprland caught my eye, and here I am :-)

workspaces here seem to work quite differently then on dwm when using multiple monitors, I have 2 monitors

```
 Monitor HDMI-A-1 (ID 0):
	1920x1080@60.000000 at 0x0
	description: Samsung Electric Company S22B150 H4MD308773 (HDMI-A-1)
	active workspace: 3 (3)
	reserved: 0 20 0 0
	scale: 1.00
	transform: 0
	focused: yes
	dpmsStatus: 1

Monitor DVI-I-1 (ID 1):
	1366x768@59.790001 at 1920x0
	description: Fujitsu Siemens Computers GmbH L19T-1 LED YV3A019380 (DVI-I-1)
	active workspace: 2 (2)
	reserved: 0 20 0 0
	scale: 1.00
	transform: 0
	focused: no
	dpmsStatus: 1
```

when i launch Hyprland, it creates workspace 1 on Monitor 1, and Workspace 2 on Monitor 2, and I'm unable to create/move Workspace 2 to monitor 1 until i switch to a different workspace on Monitor 2, this behavior is really annoying me, I have tried the following settings:

```
# Binding workspaces to a monitor
wsbind = 1, HDMI-A-1
wsbind = 2, HDMI-A-1
wsbind = 3, HDMI-A-1
wsbind = 4, HDMI-A-1
wsbind = 5, HDMI-A-1

wsbind = 6, DVI-I-1
wsbind = 7, DVI-I-1
wsbind = 8, DVI-I-1
wsbind = 9, DVI-I-1
wsbind = 0, DVI-I-1

workspace = HDMI-A-1,1
workspace = HDMI-A-1,2
workspace = HDMI-A-1,3
workspace = HDMI-A-1,4
workspace = HDMI-A-1,5

workspace=DVI-I-1,6
workspace=DVI-I-1,7
workspace=DVI-I-1,8
workspace=DVI-I-1,9
workspace=DVI-I-1,0

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, focusmonitor, HDMI-A-1
bind = $mainMod, 1, workspace, 1

bind = $mainMod, 2, focusmonitor, HDMI-A-1
bind = $mainMod, 2, workspace, 2

bind = $mainMod, 3, focusmonitor, HDMI-A-1
bind = $mainMod, 3, workspace, 3

bind = $mainMod, 4, focusmonitor, HDMI-A-1
bind = $mainMod, 4, workspace, 4

bind = $mainMod, 5, focusmonitor, HDMI-A-1
bind = $mainMod, 5, workspace, 5

bind = $mainMod, 6, focusmonitor, DVI-I-1
bind = $mainMod, 6, workspace, 6

bind = $mainMod, 7, focusmonitor, DVI-I-1
bind = $mainMod, 7, workspace, 7

bind = $mainMod, 8, focusmonitor, DVI-I-1
bind = $mainMod, 8, workspace, 8

bind = $mainMod, 9, focusmonitor, DVI-I-1
bind = $mainMod, 9, workspace, 9

bind = $mainMod, 0, focusmonitor, DVI-I-1
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspacesilent, 1
bind = $mainMod SHIFT, 2, movetoworkspacesilent, 2
bind = $mainMod SHIFT, 3, movetoworkspacesilent, 3
bind = $mainMod SHIFT, 4, movetoworkspacesilent, 4
bind = $mainMod SHIFT, 5, movetoworkspacesilent, 5
bind = $mainMod SHIFT, 6, movetoworkspacesilent, 6
bind = $mainMod SHIFT, 7, movetoworkspacesilent, 7
bind = $mainMod SHIFT, 8, movetoworkspacesilent, 8
bind = $mainMod SHIFT, 9, movetoworkspacesilent, 9
bind = $mainMod SHIFT, 0, movetoworkspacesilent, 10
```

and regardless of it, upon first launch, workspace 2 still launches on monitor 2, until i toggle workspace 6, when i toggle workspace 6 and kill the windows in Workspace 2, then I can create workspace 2 on monitor 1, is there any way around this?

also, is there a way to configure workspaces to work similarly to dwm? eg for the workspace to be created on the focused monitor, cuz sometimes i need workspace 7 on Monitor 1, and Workspace 1 on Monitor 2, or is binding workspaces to a monitor the only way to get what i want?

**Comment by vaxerski** [comment]
idk maybe reading the wiki would help who'd have thought

```
bind = $mainMod,1,moveworkspacetomonitor,1 current
bind = $mainMod,1,workspace,1
```

**Reply by kronikpillow** [reply]
wow why the attitude? maybe read the question before going full blown rude towards a new user in a unfamiliar WM, your answer doesn't answer my question, with that binding workspace 2 still starts on Monitor 2, and then I have to move it to Monitor 1, which ultimately doesn't solve the annoyance of the fact that Workspace 2 is bound to Monitor 1, but is starting on Monitor 2 until i actually start workspace 6 on Monitor 2

and before you continue to be rude, I did read the wiki, multiple times, and still am, haven't found the answer anywere, Workspace 2 always starts on Monitor 2, which is not what I want if it's already bound to Monitor 1 it should be starting on Monitor 1, while Workspace 6 should be starting on Monitor 2, since Workspace 6 7 8 9 0 are bound to Monitor 2

**Reply by vaxerski** [reply]
https://wiki.hyprland.org/Configuring/Monitors/#rotating-and-the-default-workspace

for the starting workspace id

```
workspace = HDMI-A-1,1
workspace = HDMI-A-1,2
workspace = HDMI-A-1,3
workspace = HDMI-A-1,4
workspace = HDMI-A-1,5

workspace=DVI-I-1,6
workspace=DVI-I-1,7
workspace=DVI-I-1,8
workspace=DVI-I-1,9
workspace=DVI-I-1,0
```

is invalid.

As the wiki states:
![image](https://user-images.githubusercontent.com/43317083/201087783-c42aabb3-92a1-402c-8a57-8c7dbdcaa185.png)
It doesn't "create" or "bind". It *sets* the default workspace. Your multiple keybinds are overwriting each other.

additionally,
```
workspace=DVI-I-1,0
``` 
is invalid, as ID 0 is not allowed.


Your explanation for what you want is basically "can it work like on dwm?" while I have never used dwm and have no idea what you are talking about. "for the workspace to be created on the focused monitor" says nothing, because if I move to a new workspace "hello world" (create it) it does create on my focused monitor.

**Reply by kronikpillow** [reply]
sorry English is not my native language, i am trying my best to explain :)

regarding dwm, dwm treats multi monitors that are different size as individually, eg. workspace 1 2 3 4 5 6 7 8 9 can exist on both monitors at the same time, but they are separate from each other, so I can have 18 workspaces, 9 on each monitor, instead of just 9 or 10 that share both monitors, so I can have workspace 2 on Monitor 1 and Monitor 2, showing completely different windows and i can move windows between monitors, (that was the 2nd question) .. but i necessarily don't need that functionality if i can successfully set the default workspaces ...

so, the wiki says `workspace=DP-1,1` will tell Hyprland to make the default workspace on Monitor 1 to be workspace 1, but isn't that exactly what i did with `workspace = HDMI-A-1,1` ?? hell, now i'm even more confused :(

thanks for the assist, really appreciate it as i'm really loving Hyprland

**Reply by vaxerski** [reply]
for that, see [hyprsome](https://github.com/sopa0/hyprsome)

> so, the wiki says workspace=DP-1,1 will tell Hyprland to make the default workspace on Monitor 1 to be workspace 1, but isn't that exactly what i did with workspace = HDMI-A-1,1 ?? hell, now i'm even more confused :(

No, you overwrote it later with 
```
workspace = HDMI-A-1,2
workspace = HDMI-A-1,3
workspace = HDMI-A-1,4
workspace = HDMI-A-1,5
```

making the behavior undefined

**Comment by kronikpillow** [comment]
hyprsome seems to be what i was asking for in terms of multi monitor workspaces :) thanks

regarding setting default workspace on each monitor ... I have now removed all the wsbind lines, and all the workspace lines, and only kept

`workspace=DVI-I-1,6` to try to set workspace 6 as the default workspace for Monitor 2, i restarted Hyprland by exiting, and logging in again, and Monitor 2 still starts with Workspace 2 as it's default, but as soon as I open Workspace 6 on Monitor 2, I am able to open workspace 2 on Monitor 1 ... :-/ 🤷🏻

**Reply by vaxerski** [reply]
maybe the name is wrong? check `hyprctl monitors`

**Reply by kronikpillow** [reply]
```
 Monitor HDMI-A-1 (ID 0):
	1920x1080@60.000000 at 0x0
	description: Samsung Electric Company S22B150 H4MD308773 (HDMI-A-1)
	active workspace: 3 (3)
	reserved: 0 20 0 0
	scale: 1.00
	transform: 0
	focused: yes
	dpmsStatus: 1

Monitor DVI-I-1 (ID 1):
	1366x768@59.790001 at 1920x0
	description: Fujitsu Siemens Computers GmbH L19T-1 LED YV3A019380 (DVI-I-1)
	active workspace: 2 (2)
	reserved: 0 20 0 0
	scale: 1.00
	transform: 0
	focused: no
	dpmsStatus: 1
```

The 1366x768 sized one is the one on tje right, unless that is a DVI-L-1 instead of a DVI-i-1, i dont think its wrong, i am at work now, i'll upload my config later to my repo and share it here ...

also, im using the latest release version.

Will get back to you tonight when im back from work, Thanks for the assist so far, appreciate it

**Reply by kronikpillow** [reply]
here it is ... [my hyprland config](https://gist.github.com/1c7d500bd8633ebab3dc1f22f40a60b4) ... if im seeing things correctly, now nothing is being overwritten, but i still can't get hyprland to start workspace 6 on Monitor 2 ... or am i missing something?

**Reply by vaxerski** [reply]
well now you have no workspace= keywords. No wonder.

Furthermore, as the wiki states, you need a specific monitor rule for the targeted monitor in order for these to work

**Reply by kronikpillow** [reply]
you know, for a moment there, i actually taught you were trying to help me ... but then i remembered your initial approach, where you absolutely didn't read a word i said, and just trolled me to RTFM, even thou I did RTFM, and did put out all the steps i did to try to make this work, but then I taught ok now after further comments this guy is actually trying to help me (I was wrong)..

the documentation says that:
`workspace=DVI-I-1,6`
will tell Hyprland to make the default workspace on DVI-I-1 a numbner 6 .... and your telling me now that I have no workspace keywords in the config anymore ... I have to ask you, have you actually looked at the config? or did you skip trough it, like you skipped trough my initial post?? ***HINT IT'S ON ROW 139*** 

and guess what? setting `workspace=DVI-I-1,6` doesn't set Workspace 6 as the default workspace on Monitor 2, Workspace 2 still starts as the default one, but that's what the documentation says, so hey KP, RTFM ... 

funny, there is another part of our conversation which you skipped trough, where your telling me 
> maybe the name is wrong? check `hyprctl monitors`  

even thou the result of hyprctl monitors is the first thing I posted in my initial post...

So I'm gonna say again that i'v tried the following:

```# Binding workspaces to a monitor
wsbind = 1, HDMI-A-1
wsbind = 2, HDMI-A-1
wsbind = 3, HDMI-A-1
wsbind = 4, HDMI-A-1
wsbind = 5, HDMI-A-1

wsbind = 6, DVI-I-1
wsbind = 7, DVI-I-1
wsbind = 8, DVI-I-1
wsbind = 9, DVI-I-1
wsbind = 0, DVI-I-1

workspace = HDMI-A-1,1
workspace = HDMI-A-1,2
workspace = HDMI-A-1,3
workspace = HDMI-A-1,4
workspace = HDMI-A-1,5

workspace=DVI-I-1,6
workspace=DVI-I-1,7
workspace=DVI-I-1,8
workspace=DVI-I-1,9
workspace=DVI-I-1,0
```
you told me that doesn't work, that that way i'm overwriting it ... ok so i changed it to this
```
# Binding workspaces to a monitor
wsbind = 1, HDMI-A-1
wsbind = 2, HDMI-A-1
wsbind = 3, HDMI-A-1
wsbind = 4, HDMI-A-1
wsbind = 5, HDMI-A-1

wsbind = 6, DVI-I-1
wsbind = 7, DVI-I-1
wsbind = 8, DVI-I-1
wsbind = 9, DVI-I-1
wsbind = 0, DVI-I-1

workspace = HDMI-A-1,1
workspace=DVI-I-1,6
```
Still doesn't work ...

I'v also tried all the above combinations without the wsbind lines, just with the workspace lines and all the above combinations without the workspace lines just with the wsbind lines, still doesn't work ...

then I tried deleting everything and basically reverting back to the defaulyt config, because of your confusing replies, where your essentially not helping me, and are just trolling, and I ended up only doing what the Wiki says which is:
```
workspace=DVI-I-1,6
```
***row 136 in my config gist*** it's there, plz don't tell me it isn't when I can see it & and still no luck, Workspace 2 is still the default workspace on Monitor 2... 

now ***I have to support Brodie Robertson's criticism of your documentation ... the Wiki, is crap, and crap is a understatement*** ... You have all these options listed but most don't say what they do, or say that they do something but don't do it, and some options are obviously connected to other options, but the wiki doesn't state that, as obviously something is missing to make this work as I'm doing exactly what the wiki says and it doesn't work .. or maybe it's a bug? but naah, it's just that our assuming a lot of things in the wiki are clear, when they are not...

oh also ...  at one point you say 
> you need a specific monitor rule for the targeted monitor in order for these to work

but that's not what your badly written wiki says when it comes to defining default workspace, it doesn't even mention that ... it just says

workspace=DVI-I-1,6
will tell Hyprland to make the default workspace on DVI-I-1 a number 6 ... just in different words, like you so eloquently pointed out when you RTFM'd me, even thou you didn't RTFP in the first place... 

but, don't worry, I won't ever make the mistake of coming to ask for your help again, since i'd rather burn in hell then be bullied by frustrated asshole, just for asking for help and wanting to use his project as my daily driver, I guess accumulating users is not his goal with Hyprland ...

tip: before you RTFM somebody .... maybe RTFP first? (read the fkn post)

peace out, I'm done , and I'm going back to dwm cuz the suckless crowd who is infamous for hostility, is less hostile then this asshole ... drops mic

**Reply by vaxerski** [reply]
![image](https://user-images.githubusercontent.com/43317083/203040911-cc53f27f-1b08-450e-8ed3-16d274e03e0a.png)

> but that's not what your badly written wiki says when it comes to defining default workspace, it doesn't even mention that ...

![image](https://user-images.githubusercontent.com/43317083/203041044-065c8394-5d56-4b33-a712-ec2a5a099676.png)

Sums up your entire tantrum.

**Comment by JustSimplyKyle** [comment]
![image](https://user-images.githubusercontent.com/68589851/203053026-2c87d96c-e04e-473a-9924-5cb1e3fc73b5.png)
I just need to ask something.
Does this make the default workspace of `HDMI-A-1` as `10`?(In other words, when I launch Hyprland, `HDMI-A-1` launches with workspace `10` as default)

**Reply by vaxerski** [reply]
provided you set a specific monitor rule for HDMI-A-1, yes

**Reply by JustSimplyKyle** [reply]
Ahh I get it now, this works.
![image](https://user-images.githubusercontent.com/68589851/203054794-ab909b2c-70b5-4da2-9636-efaaad247ca3.png)
But this doesn't, which is fully intentional.
![image](https://user-images.githubusercontent.com/68589851/203054998-12b2cc33-ded9-4387-8d64-53d980b01796.png)
I think an example in the wiki would be easier to grasp for new comers then
